From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Wed, 28 Oct 2020 17:29:48 +0800
Subject: cursor-tracker: Check for NULL coordinates before retrieving them

This makes Wayland consistent with X11 in not emitting warnings when
unwanted parameters are NULL.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1519
---
 src/backends/meta-cursor-tracker.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/backends/meta-cursor-tracker.c b/src/backends/meta-cursor-tracker.c
index 943949d..c6afd75 100644
--- a/src/backends/meta-cursor-tracker.c
+++ b/src/backends/meta-cursor-tracker.c
@@ -501,7 +501,9 @@ get_pointer_position_clutter (graphene_point_t *point,
   seat = clutter_backend_get_default_seat (clutter_get_default_backend ());
   cdevice = clutter_seat_get_pointer (seat);
 
-  clutter_input_device_get_coords (cdevice, NULL, point);
+  if (point)
+    clutter_input_device_get_coords (cdevice, NULL, point);
+
   if (mods)
     *mods = clutter_input_device_get_modifier_state (cdevice);
 }
